cpp/src/arrow/io/file.h

```diff
@@ -185,9 +185,14 @@ class ARROW_EXPORT MemoryMappedFile : public ReadWriteFileInterface {
   static Status Create(const std::string& path, int64_t size,
                        std::shared_ptr<MemoryMappedFile>* out);
 
+  // mmap() with whole file
   static Status Open(const std::string& path, FileMode::type mode,
                      std::shared_ptr<MemoryMappedFile>* out);
 
+  // mmap() with a region of file, the offset must be a multiple of the page size
+  static Status Open(const std::string& path, FileMode::type mode, const int64_t length,
+                     const int64_t offset, std::shared_ptr<MemoryMappedFile>* out);
```

Comment:
I would favor `(offset, length)` rather than `(length, offset)` in the signature.

Reply:
done